site/src/testHelpers/entities.ts

```diff
@@ -1649,25 +1649,36 @@ export const MockStartupLogs: TypesGen.WorkspaceAgentStartupLog[] = [
     created_at: "2023-05-04T11:30:41.402072Z",
     output: "+ curl -fsSL https://code-server.dev/install.sh",
     level: "info",
+    eof: false,
   },
   {
     id: 166664,
     created_at: "2023-05-04T11:30:41.40228Z",
     output:
       "+ sh -s -- --method=standalone --prefix=/tmp/code-server --version 4.8.3",
     level: "info",
+    eof: false,
   },
   {
     id: 166665,
     created_at: "2023-05-04T11:30:42.590731Z",
     output: "Ubuntu 22.04.2 LTS",
     level: "info",
+    eof: false,
   },
   {
     id: 166666,
     created_at: "2023-05-04T11:30:42.593686Z",
     output: "Installing v4.8.3 of the amd64 release from GitHub.",
     level: "info",
+    eof: false,
+  },
+  {
+    id: 166667,
+    created_at: "2023-05-04T11:30:42.599999Z",
+    output: "",
+    level: "info",
+    eof: true,
```

Comment:
Thinking loud:

Let's say that due to a bug we won't insert/store the EOF in the database. Will it affect the presentation layer or coderd logic?

Reply:
Nope, the frontend completely ignores EOF at the moment, but we could use it in the future to show e.g. when the log was closed. Or, perhaps which error code it exited with.

Likewise, the frontend mainly relies on the agent lifecycle state, so if the state changes the log will be hidden anyway.

We could make writing EOF in lifecycle state change mandatory to succeed, simply means the agent will keep trying to change the state until we can finally write EOF. (Currently we only warn log). That would take care of the edge case you mention.

Reply:
This is now handled in [`eb7a7a3` (#8029)](https://github.com/coder/coder/pull/8029/commits/eb7a7a366589c5d13e78e0bdc7ab7077135c2d8d). If we fail to write EOF, the agent will simply retry the lifecycle state report. An up-to-date agent will close the log file before this, though.